lib/job.js

```diff
@@ -190,10 +192,11 @@ Job.prototype.moveToFailed = function(err, ignoreLock){
     // Check if an automatic retry should be performed
     if(_this.attemptsMade < _this.opts.attempts && !_this._discarded){
       // Check if backoff is needed
-      var backoff = _this._getBackOff();
-      if(backoff){
+      if(_this.opts.backoff){
```

Comment:
in the previous implementation backoff could be 0, meaning that in this case it will be retried immediately. This differs from the new implementation where it is still sent to the delay set.

Reply:
Good spot. I changed it now to behave as before, but I am not sure how best to create a testcase for this.